### ISSUE TITLE
Username Add/Remove issues

### ISSUE THREAD

**Issue by mubaris** [curiositylab/curiosity#51]
Add new username in Settings button is not working. If I try to add a new username, it says `Username not found`.

CC: @maiquynhtruong

**Comment by maiquynhtruong** [comment]
looks like there is no quick fix. 
Why don't you request changes on my PR?

**Comment by mubaris** [comment]
We already merged those PRs. That's the reason for the issue. I think it's due to limit of API requests. You are checking for the validity of username before adding it. You need to include the Token in there. That's all.

**Comment by maiquynhtruong** [comment]
if your token is above the API request limit, shouldn't you need to get a new one?

**Comment by mubaris** [comment]
For checking, we are calling API without token.

Example:- `api.github.com/users/mubaris`

We need to change to something like this

`api.github.com/users/mubaris?access_token=${tokenVar}`

**Comment by mubaris** [comment]
I noted one more thing. Username array is still not persistent. You can add/remove usernames, but it just removes when you refresh.

**Comment by Chris-Coffman** [comment]
Should the username array be stored differently to allow for long term modification of user preferences? Perhaps storing the array in a cookie after the user sets their api token.

**Comment by mubaris** [comment]
We want to store this username array in `localStorage`

**Comment by Chris-Coffman** [comment]
But I take it you're ok will removing it from the current const array?

**Comment by mubaris** [comment]
The current const array is the default array. It's the fallback. If we are storing it in `localStorage` user can play with it. And if the user removes it from `localStorage`, we will reset `localStorage` from const array.

**Comment by Chris-Coffman** [comment]
OK, I see. What if the user wants to remove one of the defaults? Or are you content that if the users removes a default it will never be remembered after the session ends?

**Comment by mubaris** [comment]
User can remove elements of the default array, but user cannot remove the array from `localStorage`. If the user does that, we will restore it from default const arrar.

**Comment by maiquynhtruong** [comment]
@mubaris I think I know why it didn't run on yours. You might be accessing the web app through https://curiositylab.github.io/curiosity/. I tried that one too and it didn't save the username either.

Anyway I've added token to api calls and remove the line `localStorage.setItem('usernames', JSON.stringify(USERNAMES));` at `if (accessToken)`. That line resets the username array in `localStorage` even though it shouldn't happen. Hope it works this time.
